tests/Functional/src/com/microsoft/aad/adal/test/AuthenticationContextTest.java

```diff
@@ -1906,6 +1906,70 @@ private AuthenticationContext getAuthenticationContext(Context mockContext, Stri
 
         return context;
     }
+    
+    @SmallTest
+    public void testVerifyBrokerRedirectUri() throws NoSuchAlgorithmException, NoSuchPaddingException, IllegalArgumentException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException{
+    	ITokenCacheStore cache = mock(ITokenCacheStore.class);
+        final AuthenticationContext authContext = new AuthenticationContext(getContext(),
+                VALID_AUTHORITY, false, cache);
+        Class<?> c = Class.forName("com.microsoft.aad.adal.AuthenticationRequest");
+        Method m = ReflectionUtils.getTestMethod(authContext, "verifyBrokerRedirectUri", c);
+
+        //test@case valid redirect uri
+        String testRedirectUri = authContext.getRedirectUriForBroker();
+        Object authRequest = AuthenticationContextTest.createAuthenticationRequest(VALID_AUTHORITY, 
+                "resource", "clientid", testRedirectUri, "loginHint");
+        Boolean testResult = (Boolean)m.invoke(authContext, authRequest);
+        assertTrue(testResult);
+
+        //test@case broker redirect uri with invalid prefix
```

Comment:
Again, like the comment in the other code review, i want each individual test is only responsible for one thing, it makes it so clear to know what the test is testing for.